drivers/main-device.js

```diff
@@ -296,11 +296,16 @@ module.exports = class mainDevice extends Homey.Device {
             await this.setValue('measure_runmode', runModeReady, check);
 
             if (currentTemp) await this.setValue('measure_temperature', toCelsius(currentTemp), check, 10, settings.round_temp);
-            if (desiredTemp) await this.setValue('target_temperature', toCelsius(desiredTemp), check, 10, settings.round_temp);
-            // if ((targetDesiredTemp === setupParams.highRangeHigh || targetDesiredtemp === setupParams.lowRangeHigh) && desiredTemp) {
-            //     await this.setValue('target_temperature', toCelsius(desiredTemp), check, 10, settings.round_temp);
-            //     await this.onCapability_TEMPERATURE(toCelsius(desiredTemp));
-            // }
+
+            targetDesiredTemp = parseFloat(targetDesiredTemp);
+
+            // Use desiredTemp if targetDesiredTemp is out of range, targetDesiredTemp is the preferred value to use.
+            if ((targetDesiredTemp === setupParams.highRangeHigh || targetDesiredTemp == setupParams.lowRangeLow) && desiredTemp) {
+                await this.setValue('target_temperature', toCelsius(desiredTemp), check, 10, settings.round_temp);
+                await this.onCapability_TEMPERATURE(toCelsius(desiredTemp));
```

Comment:
This might cause a loop. As it will trigger `setCapabilityValues` again from `onCapability_TEMPERATURE`

Reply:
Hmm, true. Need to rethink a bit.